core/src/main/java/org/bitcoinj/script/Script.java

```diff
@@ -102,27 +104,55 @@
 
     /** Creates an empty script that serializes to nothing. */
     private Script() {
-        chunks = Lists.newArrayList();
+        this(ScriptPattern.BITCOIN);
     }
 
     // Used from ScriptBuilder.
     Script(List<ScriptChunk> chunks) {
+        this(ScriptPattern.BITCOIN, chunks);
+    }
+
+    /**
+     * Construct a Script that copies and wraps the programBytes array. The array is parsed and checked for syntactic
+     * validity.
+     * @param programBytes Array of program bytes from a transaction.
+     */
+    public Script(byte[] programBytes) throws ScriptException {
+        this(ScriptPattern.BITCOIN, programBytes);
+    }
+
+    public Script(byte[] programBytes, long creationTimeSeconds) throws ScriptException {
+        this(ScriptPattern.BITCOIN, programBytes, creationTimeSeconds);
+    }
+
+    /** Creates an empty script that serializes to nothing. */
+    private Script(ScriptPattern pattern) {
+        this.pattern = pattern;
+        chunks = Lists.newArrayList();
+    }
+
+    // Used from ScriptBuilder.
+    Script(ScriptPattern pattern, List<ScriptChunk> chunks) {
+        this.pattern = pattern;
         this.chunks = Collections.unmodifiableList(new ArrayList<>(chunks));
         creationTimeSeconds = Utils.currentTimeSeconds();
     }
 
     /**
      * Construct a Script that copies and wraps the programBytes array. The array is parsed and checked for syntactic
      * validity.
+     * @param pattern The pattern used form methods like isSentToAddress
      * @param programBytes Array of program bytes from a transaction.
      */
-    public Script(byte[] programBytes) throws ScriptException {
+    public Script(ScriptPattern pattern, byte[] programBytes) throws ScriptException {
```

Comment:
Why not call `this(pattern, programBytes, 0)`?

Reply:
Good point.